### PR TITLE
Rework mental health medication question

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

https://trello.com/c/jcrnUhy5/666-health-rework-question-are-they-prescribed-any-medication-for-their-mental-health

E2E PR: https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-e2e/pull/65

PR to update API seed data: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1358.

# Changes in this PR

Reorganises mental health medication question and follow ups.

### Before

Screenshot 2024-01-18 at 16 29 42

### After

Screenshot 2024-01-18 at 16 29 01

Screenshot 2024-01-18 at 16 29 11

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by patrickjfl** [comment]
[Health: rework question "Are they prescribed any medication for their mental health?"](https://trello.com/c/jcrnUhy5/666-health-rework-question-are-they-prescribed-any-medication-for-their-mental-health)

File: integration_tests/fixtures/applicationData.json [hunk 1]
```diff
@@ -154,9 +154,10 @@
       "isEngagedWithCommunity": "no",
       "servicesDetail": "",
       "isEngagedWithServicesInCustody": "no",
-      "hasPrescribedMedication": "no",
-      "medicationDetail": "",
-      "medicationIssues": ""
+      "canManageMedication": "notPrescribedMedication",
+      "canManageMedicationNotes": "",
```
Comment (Harriethw) [inline integration_tests/fixtures/applicationData.json:158]: minor thing, but I think if the user answers `notPrescribedMedication` then there won't be any keys in the application data for the next three fields.

File: server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.test.ts [hunk 2]
```diff
@@ -13,39 +13,6 @@ describe('MentalHealth', () => {
     })
   })
 
-  describe('Questions', () => {
```
Comment (Harriethw) [inline server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.test.ts:16]: nice

File: server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.ts [hunk 3]
```diff
@@ -104,10 +99,13 @@ export default class MentalHealth implements TaskListPage {
       delete this.body.servicesDetail
     }
 
-    if (this.body.hasPrescribedMedication !== 'yes') {
-      delete this.body.isInPossessionOfMeds
-      delete this.body.medicationDetail
+    if (this.body.canManageMedication !== 'yes') {
```
Comment (Harriethw) [inline server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.ts:102]: do we need a condition here to check if the answer is changed to the third option,`They are not prescribed medication for their mental health` ?

Comment (Harriethw) [reply server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.ts:102]: oh my bad, I realise it's covered by checking it's neither 'yes' or 'no'

File: integration_tests/fixtures/applicationData.json (hunk 1 above)
Comment (libuk) [reply integration_tests/fixtures/applicationData.json:159]: Good spot. It looks like we include empty keys in other places too, but in any event, it shouldn't give us a false-positive, so I think this is OK to go in as is.

**Review by libuk — APPROVED** [review]

**Pull request merged**